src/renderer/components/kube-object/kube-object-details.tsx

```diff
@@ -36,18 +46,18 @@ export function hideDetails() {
   showDetails();
 }
 
-export function getDetailsUrl(details: string, resetSelected = false) {
-  const detailsUrl = kubeDetailsUrlParam.toSearchString({ value: details });
+export function getDetailsUrl(selfLink: string, resetSelected = false, mergeGlobals = true) {
+  const params = new URLSearchParams(mergeGlobals ? navigation.searchParams : "");
 
-  if (resetSelected) {
-    const params = new URLSearchParams(detailsUrl);
-
-    params.delete(kubeSelectedUrlParam.name);
+  params.set(kubeDetailsUrlParam.urlName, selfLink);
 
-    return `?${params.toString()}`;
+  if (resetSelected) {
+    params.delete(kubeSelectedUrlParam.urlName);
```

Comment:
Is it a bit awkward that we set a `param` and than delete it under condition? It it possible to move first `set` under condition in some way too?

Reply:
We set _kube**Details**UrlParam_, but updating _different_ param under condition: _kube**Selected**UrlParam_.

Reply:
👍